app/controllers/nonsipp/memberpensionpayments/MemberPensionPaymentsListController.scala

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.memberpensionpayments
+
+import com.google.inject.Inject
+import config.Constants
+import config.Constants.maxNotRelevant
+import config.Refined.OneTo300
+import controllers.PSRController
+import controllers.actions._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.{refineMV, refineV}
+import forms.YesNoPageFormProvider
+import models.SchemeId.Srn
+import models._
+import navigation.Navigator
+import pages.nonsipp.memberdetails.MembersDetailsPages.MembersDetailsOps
+import pages.nonsipp.memberpensionpayments.MemberPensionPaymentsListPage
+import play.api.data.Form
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import viewmodels.DisplayMessage.{LinkMessage, Message, ParagraphMessage}
+import viewmodels.implicits._
+import viewmodels.models.{ActionTableViewModel, FormPageViewModel, PaginatedViewModel, TableElem}
+import views.html.TwoColumnsTripleAction
+
+import javax.inject.Named
+
+class MemberPensionPaymentsListController @Inject()(
+  override val messagesApi: MessagesApi,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  val controllerComponents: MessagesControllerComponents,
+  view: TwoColumnsTripleAction,
+  formProvider: YesNoPageFormProvider
+) extends PSRController {
+
+  val form = MemberPensionPaymentsListController.form(formProvider)
+
+  def onPageLoad(srn: Srn, page: Int, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) {
+    implicit request =>
+      val memberList = request.userAnswers.membersDetails(srn)
+
+      if (memberList.nonEmpty) {
+        val viewModel = MemberPensionPaymentsListController
+          .viewModel(srn, page, mode, memberList, request.userAnswers)
+        Ok(view(form, viewModel))
+      } else {
+        Redirect(
+          controllers.routes.UnauthorisedController.onPageLoad()
+        )
+      }
+  }
+
+  def onSubmit(srn: Srn, page: Int, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) { implicit request =>
+    val memberList = request.userAnswers.membersDetails(srn)
+
+    if (memberList.size > Constants.maxSchemeMembers) {
+      Redirect(
+        navigator.nextPage(MemberPensionPaymentsListPage(srn), mode, request.userAnswers)
```

Comment:
Isn't this the same as the redirection in line 87? I'm struggling to understand why the if selection is needed.